src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2031,10 +2059,22 @@ bool tryDecodeExtensionType(out bool isExplicit, [NotNullWhen(true)] out TypeSym
                         {
                             var info = new CSDiagnosticInfo(ErrorCode.ERR_MalformedExtensionInMetadata, this); // PROTOTYPE need to report use-site diagnostic
                             underlyingType = new ExtendedErrorTypeSymbol(type, LookupResultKind.NotReferencable, info, unreported: true);
+                            return false;
                         }
                         else
                         {
-                            underlyingType = type;
+                            // Validate instance field
+                            if (!underlyingInstanceField.IsNil
+                                && !validateUnderlyingInstanceField(underlyingInstanceField, moduleSymbol, type))
```

Comment:
>!validateUnderlyingInstanceField(underlyingInstanceField, moduleSymbol, type) [](http://example.com/codeflow?start=35&length=77)

It feels like field validation doesn't belong in the loop that checks parameters. Why can't we do this after the loop?